### PR TITLE
use material-ui, create public and private route, create guest page

### PR DESCRIPTION
**Pull request by AlexeyUlyanov96**

### CODE REVIEW

File: backend/routes/auth.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ router.post(
       if (!errors.isEmpty()) {
         return res
           .status(HttpStatus.BAD_REQUEST)
-          .json(new ResponseData(errors.array(), Code.ERROR));
+          .json(new ResponseData(errors.array()[0].msg, Code.ERROR));
```
Comment (avmaxim) [inline backend/routes/auth.js]: Wow. A lot of questions around this - `errors.array()[0].msg`. Let's start with the 1st one - why is it implemented like this?

File: backend/routes/auth.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ router.post(
...
```
Comment (avmaxim) [inline backend/routes/auth.js]: Same as above right here

File: backend/routes/auth.js (hunk 1 above)
Comment (AlexeyUlyanov96) [reply backend/routes/auth.js]:
```
const { msg: message } = errors.array()[0];
.json(new ResponseData(message, Code.ERROR));
```
I can do it like this

File: src/components/Routing/authService/authService.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import StorageService from '../../../utils/storageService';
+
+class AuthService {
+  constructor(storage) {
+    this.storageService = new StorageService(storage);
+  }
```
Comment (avmaxim) [inline src/components/Routing/authService/authService.js]: Let's use it as a singleton. The same for `userService`

File: src/components/Routing/authService/withAuthService.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import BrowserStorageContext from '../../Common/context/browserStorageContext';
+import AuthService from './authService';
+
+const withAuthService = Component => props => (
+  <BrowserStorageContext.Consumer>
+    {storage => <Component {...props} authService={new AuthService(storage)} />}
```
Comment (avmaxim) [inline src/components/Routing/authService/withAuthService.js]: 👍

File: src/components/Routing/appPaths.js [hunk 5]
```diff
@@ -2,7 +2,9 @@ const AppPaths = {
   HOME: '/',
   PROFILE: '/profile',
   SEARCH: '/search',
-  SIGN_UP: '/sign-up'
+  GUEST: '/guest',
```
Comment (avmaxim) [inline src/components/Routing/appPaths.js]: No need for the Guest page here.

File: src/config/apiInstance.js [hunk 6]
```diff
@@ -2,8 +2,11 @@ import * as axios from 'axios';
 import { OK, UNAUTHORIZED } from 'http-status-codes';
 import StorageService from '../utils/storageService';
 import RequestApi from '../api/api';
+import history from '../components/Routing/history';
+import AppPaths from '../components/Routing/appPaths';
 
 const BASE_URL = 'http://192.168.100.98:5000/';
+const storageService = new StorageService(localStorage);
```
Comment (avmaxim) [inline src/config/apiInstance.js]: Oh nooooo 🤯Duplicate?? 
```const storageService = new StorageService(localStorage); ```
and 
image
and 
image

File: src/components/Common/fieldData.js [hunk 7]
```diff
@@ -16,3 +17,12 @@ export const Placeholder = {
...
 };
+export const Label = {
+  [FieldName.FIRST_NAME]: 'First Name',
+  [FieldName.LAST_NAME]: 'Last Name',
+  [FieldName.EMAIL]: 'Email',
+  [FieldName.PASSWORD]: 'Password',
+  [FieldName.WEIGHT]: 'Weight',
+  [FieldName.HEIGHT]: 'Height',
+  [FieldName.BIO]: 'Biography'
+};
```
[the remark this answers is not in the document]
Comment (AlexeyUlyanov96) [reply src/components/Common/fieldData.js]: `export const Label = Placeholder;` - оставим так, с тудушкой)

File: src/components/GuestPage/styles.GuestPage.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    height: '100vh'
```
[the remark this answers is not in the document]
Comment (AlexeyUlyanov96) [reply src/components/GuestPage/styles.GuestPage.js]: it's need for `<Grid />` from material-ui
Screen Shot 2020-02-06 at 20 03 35

File: src/components/Common/context/browserStorageContext.js [hunk 9]
```diff
@@ -0,0 +1,4 @@
+import { createContext } from 'react';
+
+const BrowserStorageContext = createContext(null);
```
[the remark this answers is not in the document]
Comment (AlexeyUlyanov96) [reply src/components/Common/context/browserStorageContext.js]: okay

File: src/config/apiInstance.js (hunk 6 above)
Comment (AlexeyUlyanov96) [reply src/config/apiInstance.js]: Fixed, the instance of `StorageService` creates only in `AuthService` now

Comment (AlexeyUlyanov96) [reply src/config/apiInstance.js]: use singletone pattern for storageService class

File: src/components/Routing/authService/authService.js (hunk 3 above)
Comment (AlexeyUlyanov96) [reply src/components/Routing/authService/authService.js]: okay, used singleton pattern for storageService class

**Pull request merged**